be/src/vec/exec/vjson_scanner.cpp

```diff
@@ -0,0 +1,729 @@
+        assert_cast<ColumnVector<Int128>*>(column_ptr)->insert_data(value_ptr, 0);
+        break;
+    }
+    case TYPE_FLOAT: {
+        assert_cast<ColumnVector<Float32>*>(column_ptr)->insert_data(value_ptr, 0);
+        break;
+    }
+    case TYPE_DOUBLE: {
+        assert_cast<ColumnVector<Float64>*>(column_ptr)->insert_data(value_ptr, 0);
+        break;
+    }
+    case TYPE_CHAR: {
+        assert_cast<ColumnString*>(column_ptr)->insert_data(value_ptr, wbytes);
+        break;
+    }
+    case TYPE_VARCHAR:
+    case TYPE_STRING: {
+        assert_cast<ColumnString*>(column_ptr)->insert_data(value_ptr, wbytes);
+        break;
+    }
+    case TYPE_OBJECT: {
+        Slice slice(value_ptr, wbytes);
+        // insert_default()
+        auto* target_column = assert_cast<ColumnBitmap*>(column_ptr);
+
+        target_column->insert_default();
+        BitmapValue* pvalue = nullptr;
+        int pos = target_column->size() - 1;
+        pvalue = &target_column->get_element(pos);
+
+        if (slice.size != 0) {
+            BitmapValue value;
+            value.deserialize(slice.data);
+            *pvalue = std::move(value);
+        } else {
+            *pvalue = std::move(*reinterpret_cast<BitmapValue*>(slice.data));
+        }
+        break;
+    }
+    case TYPE_HLL: {
+        Slice slice(value_ptr, wbytes);
+        auto* target_column = assert_cast<ColumnHLL*>(column_ptr);
+
+        target_column->insert_default();
+        HyperLogLog* pvalue = nullptr;
+        int pos = target_column->size() - 1;
+        pvalue = &target_column->get_element(pos);
+        if (slice.size != 0) {
+            HyperLogLog value;
+            value.deserialize(slice);
+            *pvalue = std::move(value);
+        } else {
+            *pvalue = std::move(*reinterpret_cast<HyperLogLog*>(slice.data));
+        }
+        break;
+    }
+    case TYPE_DECIMALV2: {
+        assert_cast<ColumnDecimal<Decimal128>*>(column_ptr)
+                ->insert_data(value_ptr, 0);
+        break;
+    }
+    case TYPE_DATETIME: {
+        Slice slice(value_ptr, wbytes);
+        DateTimeValue value = *reinterpret_cast<DateTimeValue*>(slice.data);
+        VecDateTimeValue date;
+        date.convert_dt_to_vec_dt(&value);
+        assert_cast<ColumnVector<Int64>*>(column_ptr)
+                ->insert_data(reinterpret_cast<char*>(&date), 0);
+        break;
+    }
+    case TYPE_DATE: {
+        Slice slice(value_ptr, wbytes);
+        DateTimeValue value = *reinterpret_cast<DateTimeValue*>(slice.data);
+        VecDateTimeValue date;
+        date.convert_dt_to_vec_dt(&value);
+        assert_cast<ColumnVector<Int64>*>(column_ptr)
+                ->insert_data(reinterpret_cast<char*>(&date), 0);
+        break;
+    }
+    default: {
+        DCHECK(false) << "bad slot type: " << slot_desc->type();
+        break;
+    }
+    }
+    return Status::OK();
+}
+
+Status VJsonReader::_vhandle_flat_array_complex_json(std::vector<MutableColumnPtr>& columns,
```

Comment:
80% the code of `_vhandle_flat_array_complex_json` of `_vhandle_nested_complex_json` is same, you shoud reuse it.